bindings/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2022 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+if(ICUB_MODELS_COMPILE_PYTHON_BINDINGS)
+
+    find_package(pybind11 REQUIRED)
+    find_package(Python3 COMPONENTS Interpreter REQUIRED)
+
+
+    # define new line accordingly to the operating system
+    if (WIN32)
+       set(NEW_LINE "\n\r")
+    else()
+       set(NEW_LINE "\n")
+    endif()
+
+    option(ICUB_MODELS_DETECT_ACTIVE_PYTHON_SITEPACKAGES
+        "Do you want icub-models to detect and use the active site-package directory? (it could be a system dir)"
+        FALSE)
+
+    # Install the resulting Python package for the active interpreter
+    if(ICUB_MODELS_DETECT_ACTIVE_PYTHON_SITEPACKAGES)
+        set(PYTHON_INSTDIR ${Python3_SITELIB}/icub_models)
+    else()
+     execute_process(COMMAND ${Python3_EXECUTABLE} -c "from distutils import sysconfig; print(sysconfig.get_python_lib(1,0,prefix=''))"
+       OUTPUT_VARIABLE _PYTHON_INSTDIR)
+
+     string(STRIP ${_PYTHON_INSTDIR} _PYTHON_INSTDIR_CLEAN)
+     set(PYTHON_INSTDIR ${_PYTHON_INSTDIR_CLEAN}/icub_models)
+    endif()
+
+    # Folder of the Python package within the build tree.
+    # It is used for the Python tests.
+    set(ICUB_MODELS_PYTHON_PACKAGE "${CMAKE_BINARY_DIR}/icub_models")
+
+    # Add the bindings directory
+    add_subdirectory(python)
+
+    # Create the __init__.py file
+    file(GENERATE
+         OUTPUT "${ICUB_MODELS_PYTHON_PACKAGE}/__init__.py"
+         CONTENT "from icub_models.bindings import *")
+
+    # Install the __init__.py file
+    install(FILES "${ICUB_MODELS_PYTHON_PACKAGE}/__init__.py"
+            DESTINATION ${PYTHON_INSTDIR})
+
+    # Install pip metadata files to ensure that blf installed via CMake is listed by pip list
+    # See https://packaging.python.org/specifications/recording-installed-packages/
+    # and https://packaging.python.org/en/latest/specifications/core-metadata/#core-metadata
+    option(ICUB_MODELS_PYTHON_PIP_METADATA_INSTALL "Use CMake to install Python pip metadata. Set to off if some other tool already installs it." ON)
+    mark_as_advanced(ICUB_MODELS_PYTHON_PIP_METADATA_INSTALL)
+    set(ICUB_MODELS_PYTHON_PIP_METADATA_INSTALLER "cmake" CACHE STRING "Specify the string to identify the pip Installer. Default: cmake, change this if you are using another tool.")
+    mark_as_advanced(ICUB_MODELS_PYTHON_PIP_METADATA_INSTALLER)
+    if(ICUB_MODELS_PYTHON_PIP_METADATA_INSTALL)
+      get_filename_component(PYTHON_METADATA_PARENT_DIR ${PYTHON_INSTDIR} DIRECTORY)
+      if(WIN32)
+        set(NEW_LINE "\n\r")
+      else()
+        set(NEW_LINE "\n")
+      endif()
+      file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/METADATA "")
+      file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/METADATA "Metadata-Version: 2.1${NEW_LINE}")
+      file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/METADATA "Name: icub_models${NEW_LINE}")
```

Comment:
I think that for pep-0008 the use of underscore in package names is discouraged, see https://www.python.org/dev/peps/pep-0008/#package-and-module-names, while the underscore is actuall suggested for the module name.

Reply:
Double checking with @diegoferigo .

Reply:
If I interpret the PEP correctly, it states that the usage of underscores in _module_ names is discouraged. Practically, if you cannot find a single name descriptive enough, this suggestion from a very old PEP is not really followed and you can find pretty often `_`.

This being said, the module name is different from the *package* name, and this section of CMake code is about the resulting package name. In this case, it does not matter much, because underscores in the metadata are parsed as dashes by PyPI and pip. What differs is just the sdist / wheel package name. And I mean, a package with name `icub-models` containing a module stored in folder `icub_models` so that it can be imported as `import icub_models`, when packaged, produces `icub-models-....{zip|wheel}`. This bother mostly CI since you have to take care of two different names. Since packages named `icub_models` can be installed with `pip` as `pip install icub-models` regardless, I'd suggest to keep all naming consistent, using always `_`.